x-pack/plugins/observability_solution/apm/public/components/app/entities/entity_link/index.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiEmptyPrompt, EuiLink, EuiLoadingSpinner, EuiImage } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { dashboardsDark, dashboardsLight } from '@kbn/shared-svg';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { ENVIRONMENT_ALL_VALUE } from '../../../../../common/environment_filter_values';
+import { useServiceEntitySummaryFetcher } from '../../../../context/apm_service/use_service_entity_summary_fetcher';
+import { useEntityManagerEnablementContext } from '../../../../context/entity_manager_context/use_entity_manager_enablement_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { FETCH_STATUS, isPending, useFetcher } from '../../../../hooks/use_fetcher';
+import { useApmRouter } from '../../../../hooks/use_apm_router';
+import { useTheme } from '../../../../hooks/use_theme';
+import { ApmPluginStartDeps } from '../../../../plugin';
+
+const pageHeader = {
+  pageTitle: 'APM',
+};
+
+export function EntityLink() {
+  const router = useApmRouter({ prependBasePath: false });
+  const theme = useTheme();
+  const { services } = useKibana<ApmPluginStartDeps>();
+  const { observabilityShared, data } = services;
+  const timeRange = data.query.timefilter.timefilter.getTime();
+  const ObservabilityPageTemplate = observabilityShared.navigation.PageTemplate;
+  const {
+    path: { serviceName },
+    query: { rangeFrom = timeRange.from, rangeTo = timeRange.to },
+  } = useApmParams('/link-to/entity/{serviceName}');
+  const { isEntityCentricExperienceViewEnabled, entityManagerEnablementStatus } =
+    useEntityManagerEnablementContext();
+  const { serviceEntitySummary, serviceEntitySummaryStatus } = useServiceEntitySummaryFetcher({
+    serviceName,
+    environment: ENVIRONMENT_ALL_VALUE,
+  });
+
+  const { data: hasApmData, status: hasApmDataStatus } = useFetcher((callApmApi) => {
+    return callApmApi('GET /internal/apm/has_data');
+  }, []);
+
+  if (
+    isPending(entityManagerEnablementStatus) ||
+    serviceEntitySummaryStatus === FETCH_STATUS.LOADING ||
+    isPending(hasApmDataStatus)
+  ) {
+    return (
+      <div>
+        <EuiLoadingSpinner />
+      </div>
+    );
+  }
+
+  if (
+    // When EEM is disabled we'll show the APM overview page.
```

Comment:
it seems that we want to redirect to the overview page in all cases expect when EEM is enabled but it doesn't have any data yet.  (Unless I miss something_

if that's the case, we could simplify this by flipping the logic and check whether the entity has data or not with the summary endpoint when `isEntityCentricExperienceViewEnabled` is true.
 
Also `GET /internal/apm/has_data'` checks for ANY data comparison to the summary endpoint for the entities.

Reply:
what do you think?

Reply:
sounds good to me

Reply:
These will be the conditions to show the EEM callout:
```
isEntityCentricExperienceViewEnabled === true **AND**
serviceEntitySummary.dataStreamTypes.length === 0 **AND**
hasApmData?.hasData === false
```

Otherwise to forward to Service overview page.